volmdlr/wires.py

```diff
@@ -1203,13 +1358,32 @@ def random_point_inside(self):
     #     print('primitives points :', [(prim[0], prim[1]) for prim in self.primitives])
 
     def order_contour(self):
+        
         if len(self.primitives) < 2:
             return self
+        
+        initial_points = []
+        for p in self.primitives:
+            initial_points.append((p.start, p.end))
+        
         new_primitives = []
         points = self.ordering_contour()
         for p1, p2 in points:
-            if p1 != p2:
+            try:
+                index = initial_points.index((p1, p2))
+            except ValueError:
+                index = initial_points.index((p2, p1))
+            
+            if type(self.primitives[index]) == volmdlr.edges.LineSegment2D:
                 new_primitives.append(volmdlr.edges.LineSegment2D(p1, p2))
+            elif type(self.primitives[index]) == volmdlr.edges.Arc2D:
+                new_primitives.append(volmdlr.edges.Arc2D(p1, self.primitives[index].interior, p2))
+            elif type(self.primitives[index]) == volmdlr.edges.BSplineCurve2D:
+                if (p1, p2) == (self.primitives[index].start, self.primitives[index].end):
+                    new_primitives.append(self.primitives[index])
+                else:
+                    new_primitives.append(self.primitives[index].reverse())
```

Comment:
The use of isinstance might be preferred here.

Reply:
It's done!